### ISSUE TITLE
Encapsulate gesture and tracking mode behaviour 

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-gl-native#6557]
**tldr**: Current system around gestures and tracking modes introduces state management in the code (ie.  if-else checks). Let's refactor this out. 
#### Gestures

The Android SDK contains zoom, scroll, rotate and tilt gestures. These items can be enabled/disabled by users through xml and code and introduce a first layer of state management.
#### Tracking settings

Tracking settings allow the user to track location/heading. Users can configure tracking settings that impacts certain gestures (eg. setting bearing gps, disables rotate gesture). On top of that we also allow to hook into gesture code that results in disabling tracking modes.
#### Proposal

We have a complex system around gestures and tracking modes. It now introduces boilerplate and is a source of bugs (ref. https://github.com/mapbox/mapbox-gl-native/issues/6549). Proposal is to look into refactoring this out into a design pattern and make it more testable.

**Comment by tobrun** [comment]
Happened with https://github.com/mapbox/mapbox-gl-native/pull/7189

**Comment by tobrun** [comment]
additional implementation notes here is that we now have an Interface that is responsible for changing the focal point -> FocalPointChangeListener.
